Gems/LyShine/Code/Source/LyShineFeatureProcessor.cpp

```diff
@@ -71,7 +71,7 @@ namespace LyShine
             //If DepthPrePass is missing then find the depth attachment via ForwardPass
             if (findIter != passRequestCopy.m_connections.end())
             {
-                if (!renderPipeline->FindFirstPass(AZ::Name("ForwardPass")))
+                if (renderPipeline->FindFirstPass(AZ::Name("ForwardPass")))
```

Comment:
If a "ForwardPass" must be found to do any operation, maybe it'd be better to to this search first and then inside the `if` do all the depth attachment stuff.

Reply:
It's not important, but nothing seems to be modifying `renderPipeline` during these checks, so the following code should be the same:

````
if (!renderPipeline->FindFirstPass(AZ::Name("DepthPrePass")))
{
    //If DepthPrePass is missing then find the depth attachment via ForwardPass
    if (renderPipeline->FindFirstPass(AZ::Name("ForwardPass")))
    {
        const AZ::Name depthAttachment = AZ::Name("DepthInputOutput");
        auto findIter = AZStd::find_if(passRequestCopy.m_connections.begin(), passRequestCopy.m_connections.end(), [depthAttachment](const AZ::RPI::PassConnection& entry)
        {
            return entry.m_localSlot == depthAttachment;
        });

        if (findIter != passRequestCopy.m_connections.end())
        {
            (*findIter).m_attachmentRef.m_pass = "ForwardPass";
            (*findIter).m_attachmentRef.m_attachment = "DepthStencilOutput";
        }
    }
    passRequest = &passRequestCopy;
}
````

Reply:
Ahh you meant to pull it above the find_if and not all the way up. Ok I can make this change since AR failed anyways.